moler/cmd/unix/sftp.py

```diff
@@ -0,0 +1,267 @@
+# -*- coding: utf-8 -*-
+"""
+SFTP command module.
+"""
+__author__ = 'Agnieszka Bylica'
+__copyright__ = 'Copyright (C) 2018, Nokia'
+__email__ = 'agnieszka.bylica@nokia.com'
+
+
+from moler.cmd.unix.genericunix import GenericUnixCommand
+from moler.exceptions import CommandFailure
+from moler.exceptions import ParsingDone
+import re
+
+
+class Sftp(GenericUnixCommand):
+    def __init__(self, connection, host, password, user=None, confirm_connection=True, pathname=None,
+                 new_pathname=None, options=None, command=None, prompt=None, new_line_chars=None):
+        super(Sftp, self).__init__(connection=connection, prompt=prompt, new_line_chars=new_line_chars)
+
+        self.host = host
+        self.user = user
+        self.password = password
+        self.confirm_connection = confirm_connection
+        self.ready_to_parse_line = False
+
+        self.pathname = pathname
+        self.new_pathname = new_pathname
+
+        self.options = options
+        self.command = command
+
+        self.command_sent = False
+        self.sending_started = False
+
+        self.current_ret['RESULT'] = list()
+
+    def build_command_string(self):
+        cmd = "sftp"
+        if self.options:
+            cmd = "{} {}".format(cmd, self.options)
+        if self.user:
+            cmd = "{} {}@{}".format(cmd, self.user, self.host)
+        else:
+            cmd = "{} {}".format(cmd, self.host)
+        if self.pathname:
+            cmd = "{}:{}".format(cmd, self.pathname)
+        if self.new_pathname:
```

Comment:
Does it work without paths?

Reply:
Yes, if there is no source path, sftp connects to server and user gets access to remote sftp terminal. Without destination path sftp downloads file to current directory.